pkg/cvo/sync_worker.go

```diff
@@ -483,7 +483,15 @@ func (w *SyncWorker) Update(ctx context.Context, generation int64, desired confi
 		return w.status.DeepCopy()
 	}
 
-	work.Capabilities = capability.SetCapabilities(config, priorCaps, capability.GetCapabilitiesAsMap(w.alwaysEnableCapabilities))
+	// TODO: use slices.Collect(maps.Keys(priorCaps)) after bumping to go1.23
+	collectKeys := func(caps map[configv1.ClusterVersionCapability]struct{}) []configv1.ClusterVersionCapability {
+		var ret []configv1.ClusterVersionCapability
+		for k := range caps {
+			ret = append(ret, k)
+		}
+		return ret
+	}
+	work.Capabilities = capability.SetCapabilities(config, append(collectKeys(priorCaps), w.alwaysEnableCapabilities...))
```

Comment:
I think this place deserves a hint about what the union actually means. The hint can be just a well-named temporary variable but at this place it is not clear what do we actually pass to the method:

```
ensureCapabilities = append(collectKeys(priorCaps), w.alwaysEnableCapabilities...)
work.Capabilities = capability.SetCapabilities(config, ensureCapabilities)
```

I'm not saying `ensureCapabilities` is a good name, just illustrates my intent. Maybe consider even a comment saying why we are enabling this union.

Reply:
Add a var `ensureEnabledCapabilities`.
My understanding is that `collectKeys(priorCaps)` is from `cv.spec` and `w.alwaysEnableCapabilities` is from the program args that could introduce some implicitly enabled caps. Implicitly enabled caps are the caps that are not in the CV's `spec.capabilities` (otherwise, the caps are explicitly enabled. I feel "explicitly" is hidden in the existing code naming convention).

https://github.com/openshift/cluster-version-operator/blob/ade334a2df6341b373f2b27f22fbdd80e490e2c6/cmd/start.go#L50

So the union is the caps that we need to ensure enabled (either explicitly or implicitly).